File: Year 1/MAP/JavaRepo - Toy language/TestExample3/src/model/ConstExp.java
package model;

public class ConstExp implements Exp{
    private int nr;

    public ConstExp(int _nr){
        nr = _nr;
    }

    @Override
    public int eval(){
        return nr;
    }

    @Override
    public String toString(){
        return "" + nr;
    }
}

File: Year 2/Semester 1/MAP/C# - Toy language/Lab10/Toy Language/Toy Language/model/expressions/ConstExp.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Toy_Language.utils;

namespace Toy_Language.model.expressions
{
    public class ConstExp : Exp
    {
        private int number;
        
        public ConstExp(int _number)
        {
            this.number = _number;
        }

        public override int Eval(IModDictionary<string, int> symbTable)
        {
            return number;
        }

        public override string ToString()
        {
            return "" + number;
        }
    }
}
